### ISSUE TITLE
Student projects roadmap summer 2016

### ISSUE THREAD

**Issue by johndbritton** [education/classroom#572]
We're expecting to have a few students focused on improving Classroom this summer with help from @tarebyte and me. Let's use this issue to plan goals for the summer and track progress against them. For a first draft, I've split tasks into three main categories: setting up, monitoring progress, and winding down. By splitting our goals across the different parts of the usage cycle my hope is that we can work on the various projects in parallel.

**Setting up (@dinever)**
- [ ] Student identifiers #614 
- [ ] Team management
- [ ] Pre-defined teams
- [ ] Improved invitation experience

**Montoring progress (@cyhsutw)**
- [ ] Repository creation
- [ ] Due dates
- [ ] Explicit submission
- [ ] Visualize classroom activity

**Winding down (@nicktikhonov)**
- [ ] Download all student submissions
- [ ] Access "submitted" commits
- [ ] Archive submissions
- [ ] Track feedback to students
- [ ] Transfer code to student ownership

I'm open to other ways of organizing this work, so please comment here if you have ideas. Also, if there's something you don't see listed here that you think we need to address, please share. I can't promise that we'll be able to implement every feature requested but my hope is to get through the most needed features in time for the fall semester.

**Comment by cyhsutw** [comment]
Hi @johndbritton @tarebyte 

Thank you so much for considering my proposal and giving me the opportunity to work on GitHub Classroom this summer.

I have some ideas/questions about the items listed above, is there any preferred way for the discussion?

**Comment by tarebyte** [comment]
I would raise another issue for a discussion thread and reference this issue.
